### PR TITLE
Server Side Rendering

### PR DESCRIPTION
**Pull request by samasastudio**
- bundled server using webpack to serve JSX syntax
- created express route to call ReactDOM.renderString and compile a component template to render with page load
- added axios call to retrieve data on component, format it and inject in Component props all on server side
- added html script tag that loads data on global window as a fail safe
- created new dev scripts for using nodemon to start bundle
- added Redis caching for page load speeds
- mobile performance scoring 70 to 83.  First content paint and index were taken up to the 2s mark and time to interactivity was taken into the green from 4.3s to 3.5s.
- ran page load stress testing with page responding at a slowest average of 0.437ms.
- was able to get artillery to run over 11,000 requests in 10s and page responded by completing up to 1184.66 requests per second

Screen Shot 2020-09-27 at 11 28 05 PM
Screen Shot 2020-09-27 at 6 45 44 PM
Screen Shot 2020-09-27 at 6 46 16 PM
Screen Shot 2020-09-27 at 8 13 27 PM

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <!-- <link href='style.css' rel='stylesheet' type='text/css'> -->
   </head>
   <body>
-    <div id='description'></div>
-    <script src="./bundle.js"></script>
+    <div id="description"></div>
```
Comment (WDHudson) [inline client/public/index.html:10]: Any reason for double quotes from single quotes?

File: client/src/render.jsx [hunk 2]
```diff
@@ -1,3 +1,5 @@
 import DescriptionService from './index.jsx';
+import React from 'react';
+import ReactDOM from 'react-dom';
 
-ReactDOM.render(<DescriptionService />, document.getElementById('description'));
\ No newline at end of file
+ReactDOM.hydrate(<DescriptionService initData={window.__initData__} initCurrent={'descriptionB'}/>, document.getElementById("description"));
```
Comment (WDHudson) [inline client/src/render.jsx:5]: Interesting. Does hydrating still render? I need to look into this.

**Review by WDHudson — APPROVED** [review]
Very interesting work! I asked a few questions just more for my curiosity instead of questioning your thought process.

File: client/public/index.html (hunk 1 above)
Comment (samasastudio) [reply client/public/index.html:10]: in the beginning i was using a method to pull my index.html using fs.fileRead and replace parts of the string with my root components for rendering.  to use string.replace() it required that all the single and double quotes be uniform, but I later changed this method later to the current setup.  overall: it doesn't matter much anymore lol, but beware of it if you use the string.replace() method.

File: client/src/render.jsx (hunk 2 above)
Comment (samasastudio) [reply client/src/render.jsx:5]: it does!  it's used specifically to hydrate server side rendered html roots with data and play nicely with ReactDOMServer.  here: https://reactjs.org/docs/react-dom.html#hydrate

**Pull request merged**